kyaml/kio/byteio_writer.go

```diff
@@ -72,15 +72,8 @@ func (w ByteWriter) Write(nodes []*yaml.RNode) error {
 			}
 		}
 
-		// TODO(pwittrock): factor this into a a common module for pruning empty values
-		_, err := nodes[i].Pipe(yaml.Lookup("metadata"), yaml.FieldClearer{
-			Name: "annotations", IfEmpty: true})
-		if err != nil {
-			return errors.Wrap(err)
-		}
-		_, err = nodes[i].Pipe(yaml.FieldClearer{Name: "metadata", IfEmpty: true})
-		if err != nil {
-			return errors.Wrap(err)
+		if err := yaml.ClearEmptyAnnotations(nodes[i]); err != nil {
```

Comment:
So does this mean we will remove the empty annotation field whenever we process a manifest?

Reply:
Yes.

Reply:
So it's not obvious to me that we should remove empty fields, since having those in the manifest is perfectly legal. Although I can't think of any situations this is useful for k8s manifests. But looks like we are already doing this, so this doesn't change the behavior.